apps/rush-lib/src/api/Rush.ts

```diff
@@ -61,10 +61,14 @@ export class Rush {
       return;
     }
 
-    const parser: RushCommandLineParser = new RushCommandLineParser({
-      alreadyReportedNodeTooNewError: options.alreadyReportedNodeTooNewError
-    });
-    parser.execute().catch(console.error); // CommandLineParser.execute() should never reject the promise
+    // Use a dynamic import here to avoid an issue where a class's (RushInstallManager) base class (BaseInstallManager)
+    // isn't defined at its class declaration because of a circular import dependency
+    import('../cli/RushCommandLineParser').then(({ RushCommandLineParser }) => {
```

Comment:
There should be a synchronous solution for this problem.

Reply:
Actually are you sure this problem still exists? When I reverted this change, it seemed to work fine.

Reply:
I can double check when I port this over. As far as I'm aware it is still needed